targets/mbedos5/tools/js2c.py

```diff
@@ -0,0 +1,276 @@
+
+            if len(js_target_list) > 1:
+                print("ERROR: Found multiple targets with the provided MCU name.")
+                print("Found targets:")
+                print('\n'.join(js_target_list))
+                sys.exit(1)
+            else:
+                os.system("cp {} ./js/pins.js".format(js_target_list[0]))
+        except:
+            print("ERROR: Cannot find pins.js file for provided MCU.")
+            sys.exit(1)    
+    else:
+        cmd = "find {} -iname \"{}\"".format(SRC_PATH + "pin_defs", "TARGET_" + args.mcu)
+        intended_target = subprocess.check_output(cmd, shell = True).strip()
+
+        if intended_target == "":
+            cmd = "find {} -iname \"{}\"".format(SRC_PATH + "pin_defs", "TARGET_MCU_" + args.mcu)
+            intended_target = subprocess.check_output(cmd, shell=True).strip()
+
+        if intended_target == "":
+            print("ERROR: Cannot find pins.js file for provided MCU.")
+            sys.exit(1)
+
+        if "\n" in intended_target:
+            print("ERROR: Found multiple targets with the provided MCU name.")
+            print("Found targets:")
+            print(intended_target)
+            sys.exit(1)
+
+        # check for multiple variants
+        cmd = "find {} -iname pins.js".format(intended_target)
+        js_files = subprocess.check_output(cmd, shell=True).strip().split('\n')
+
+        if len(js_files) > 1:
+            # found multiple
+            js_files = [js_file for js_file in js_files if args.variant in js_file]
+
+        if len(js_files) > 1:
+            print("ERROR: Found multiple pins.js files with the provided MCU name.")
+            print("Found js files:")
+            print(js_files)
+            sys.exit(1)
+
+        os.system("cp {} js/pins.js".format(js_files[0]))
+    print("Updated pins.js")
+
+jshint_is_installed = False
+try:
+    with open(os.devnull, "w") as dev_null:
+        if platform.system() == "Windows":
+            subprocess.check_call("where jshint", shell=True, stdout=dev_null)
+        else:
+            subprocess.check_call("which jshint", shell=True, stdout=dev_null)
+        jshint_is_installed = True
+except:
+    print("No JSHint detected, cannot statically check if PinNames you use are valid for your platform.")
+    print("Consider installing JSHint for this feature.")
+
+if jshint_is_installed and platform.system() != "Windows":
+    try:
+        subprocess.check_call("tools/check_pins.sh")
+    except:
+        print("JSHint picked up an undefined variable in your code.")
+        print("Are you using a PinName that doesn't exist for your platform?")
+        raise
+        sys.exit(1)
+
+fout = open(OUT_PATH + 'js_encoded.cpp', 'w')
+fout.write(LICENSE);
+fout.write(HEADER);
+
+pins = list()
+pin_def_re = re.compile("var ([a-zA-Z_0-9]+)\s*=\s*([^;]*);")
+
+def exportOneFile(path, name):
+    # Prepend pins.js onto main.js, don't include pins.js itself.
+    code = ""
+    if name == "pins":
```

Comment:
I would not put the pin name/value extraction here as it clearly not the same as the methods original purpose. Simply add a check where the method is called and do the extraction there (preferably as a method).